pkg/apis/aws/validation/worker.go

```diff
@@ -71,17 +72,18 @@ func ValidateWorkerConfig(workerConfig *apisaws.WorkerConfig, volume *core.Volum
 	}
 
 	if nodeTemplate := workerConfig.NodeTemplate; nodeTemplate != nil {
-		if len(nodeTemplate.Capacity) == 0 {
-			allErrs = append(allErrs, field.Required(fldPath.Child("nodeTemplate").Child("capacity"), "capacity must not be empty"))
-		}
-		for _, capacityAttribute := range []corev1.ResourceName{"cpu", "gpu", "memory"} {
-			value, ok := nodeTemplate.Capacity[capacityAttribute]
-			if !ok {
-				// core resources such as "cpu", "gpu", "memory" need not all be explicitly specified in workerConfig.NodeTemplate.
-				// Will fall back to the worker pool's node template if missing.
-				continue
+		for capacityAttribute, value := range nodeTemplate.Capacity {
+			// core resources such as "cpu", "gpu", "memory" need not all be explicitly specified in workerConfig.NodeTemplate.
+			// Will fall back to the worker pool's node template if missing.
+			if capacityAttribute == corev1.ResourceCPU { // CPU is allowed to be a decimal value, such as 1500m.
+				allErrs = append(allErrs, validateResourceQuantityValue(capacityAttribute, value, fldPath.Child("nodeTemplate").Child("capacity").Child(string(capacityAttribute)))...)
+			} else { // extended resources need to be whole, non-negative numbers. GPU and memory also need to be whole numbers.
+				allErrs = append(allErrs, validateResourceQuantityWholeNumber(capacityAttribute, value, fldPath.Child("nodeTemplate").Child("capacity").Child(string(capacityAttribute)))...)
```

Comment:
I would really appreciate a comment with the link to the ext.resources page because for sure we will forget the integer only restriction. 

Also I am not sure if there is any point in advertising a resource of `0` value. Is it something that we should maybe validate against ?